src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -222,15 +232,82 @@ protected void runCommand(ExecutorDriver driver, Task task, String command) {
     }
   }
 
+  protected boolean unhealthy(ExecutorDriver driver, Task task) {
+    log.info("Performing health check for task: " + task.taskInfo.getTaskId().getValue());
+
+    Process healthCmd = null;
+    String nodeName = null;
+    String healthCheckCmd = "netstat -plnat | grep ";
+    // TODO this code is working, but it is a mess/ refactor and use utility methods
```

Comment:
As mentioned in my TODO, can we break this method down and not hard code the ports. We can make them constants.